File: test-ui/src/main/resources/static/app/services/user.service.js
'use strict';

angular.module('test').service('UserService', ['$http', function ($http) {
    
    var service = {};

    const API_URI = 'http://localhost:5000/api'

    service.findAll = function () {        
        return $http.get(`${API_URI}/user`);        
    }

    service.saveNewUser = function (user) {
        return $http.post(`${API_URI}/user`,user);
    }
    
    service.removeUser = function(userId) {
    	return $http.delete(`${API_URI}/user/${userId}`);
    }
    
    service.findById = function (userId) {
    	return $http.get(`${API_URI}/user/${userId}`);
    }
    
    service.updateUser = function (user) {
    	return $http.put(`${API_URI}/user/${user.id}`, user);
    }

    return service;
}])

File: test-service/src/main/java/com/test/service/api/services/UserService.java
package com.test.service.api.services;

import java.util.List;

import com.test.model.api.models.User;

public interface UserService {

	public User insert(User user);	
	public List<User> findAll();
	public User findById(Integer id);
	public void deleteById(Integer id);
	public User updateUser(User user);
	
}
